script.js

```diff
@@ -0,0 +1,245 @@
+        task.isDone ? completedTasksBlock.append(taskLayout) : currentTasksBlock.append(taskLayout);
+    } else {
+        task.isDone ? completedTasksBlock.prepend(taskLayout) : currentTasksBlock.prepend(taskLayout);
+    }
+}
+
+function renderApp() {
+    currentTasksBlock.innerHTML = null;
+    completedTasksBlock.innerHTML = null;
+    allTasks.map(task => addTaskLayout(task));
+    const tasksCount = getTasksCount();
+    currentTasksTitle.textContent = `ToDo (${tasksCount.current})`;
+    completedTasksTitle.textContent = `Completed (${tasksCount.completed})`;
+}
+
+function removeTask(taskId) {
+    const copyTasks = allTasks.filter(task => task.id !== taskId);
+    updateLocalStorage(LocalStorageKeys.tasks, copyTasks);
+    allTasks = copyTasks;
+    renderApp();
+}
+
+function changeTaskStatus(taskId) {
+    const copyTasks = allTasks.map(task => task.id === taskId ? {...task, isDone: !task.isDone} : task);
+    updateLocalStorage(LocalStorageKeys.tasks, copyTasks);
+    allTasks = copyTasks;
+    renderApp();
+}
+
+function showEditModal(taskId) {
+    const task = allTasks.find(task => task.id === taskId);
+    modalLabelText.textContent = KeysVariables.editTask;
+    submitButton.textContent = KeysVariables.save;
+    inputTitle.value = task.title;
+    inputText.value = task.text;
+    form.elements[task.priority].checked = true;
+    editTaskId = taskId;
+    modalEditMode = true;
+}
+
+
+function tasksBlockListener({target}) {
+    const dropDownMenu = target.closest('.dropdown-menu');
+    if (dropDownMenu) {
+        const taskId = dropDownMenu.getAttribute('id');
+        if (target.classList.contains(KeysVariables.editBtn)) {
+            showEditModal(taskId);
+        } else if (target.classList.contains(KeysVariables.changeStatusBtn)) {
+            changeTaskStatus(taskId);
+        } else {
+            removeTask(taskId);
+        }
+    }
+}
+
+currentTasksBlock.addEventListener('click', tasksBlockListener);
+completedTasksBlock.addEventListener('click', tasksBlockListener);
+
+function closeModal() {
+    modalLabelText.textContent = KeysVariables.addTask;
+    submitButton.textContent = KeysVariables.addTask;
+    form.querySelectorAll('.form-check-input')[1].checked = true;
+    inputTitle.value = '';
+    inputText.value = '';
+    modalEditMode = false;
+}
+
+function getSelectedPriority() {
+    form.querySelectorAll('.form-check-input').forEach((checkbox) => {
+        if (checkbox.checked) {
+            return checkbox.value;
+        }
+    })
+    return null;
+}
+
+function editTask() {
+    const task = allTasks.find(task => task.id === editTaskId);
+    task.priority = getSelectedPriority();
+    task.title = inputTitle.value;
+    task.text = inputText.value;
+}
+
+function getTimeValues(time) {
+    const date = new Date(time);
+    const timeString = date.toTimeString().split('').splice(0, 5).join('');
+    const dateString = date.toLocaleDateString();
+    return {timeString, dateString};
+}
```

Comment:
It's not good approach in this case. Split it in 2 helper methods.

Reply:
Replaced to 2 functions.